### ISSUE TITLE
Responsive Razor Pages not working

### ISSUE THREAD

**Issue by wangkanai** [wangkanai/Detection#350]
From PR #349 detection sample, let me see again that razor pages is not being responsive to device.

![image](https://user-images.githubusercontent.com/10666633/74602690-0ed4a480-50de-11ea-95b9-181e2a99b793.png)

Question, how come my unit test is not catching this? 😒

**Comment by wangkanai** [comment]
![image](https://user-images.githubusercontent.com/10666633/74602721-77bc1c80-50de-11ea-883a-9ae38223662f.png)

**Comment by wangkanai** [comment]
Identified that view location expander is causing this issue.
```c#
private static IServiceCollection AddRazorViewLocation(this IServiceCollection services)
    => services.Configure<RazorViewEngineOptions>(options =>
    {
        //options.ViewLocationExpanders.Add(new ResponsiveViewLocationExpander(ResponsiveViewLocationFormat.Suffix));
        //options.ViewLocationExpanders.Add(new ResponsiveViewLocationExpander(ResponsiveViewLocationFormat.Subfolder));
        options.ViewLocationExpanders.Add(new ResponsivePageLocationExpander());
    });
```
![image](https://user-images.githubusercontent.com/10666633/74602765-eac59300-50de-11ea-85c6-cd3dbea8c762.png)

**Comment by wangkanai** [comment]
The issue is back again?

![image](https://user-images.githubusercontent.com/10666633/74620496-eabfa480-516b-11ea-897f-3d7bdd36f082.png)

**Comment by wangkanai** [comment]
MVC and Razor Pages can't not co-exist together?

Razor Page Only
![image](https://user-images.githubusercontent.com/10666633/74624083-afc46d80-5179-11ea-964f-45c03da84591.png)

**Comment by wangkanai** [comment]
Found it!!  `app.UseDetection()` must be before `app.UseRouting()`.

**Before - Failed**
```c#
public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
{
    app.UseStaticFiles();

    app.UseDetection();
    
    app.UseRouting();

    app.UseEndpoints(endpoints => { endpoints.MapRazorPages(); });
}
```
![image](https://user-images.githubusercontent.com/10666633/74633477-4ef65e80-5194-11ea-88fe-2af715cc1d98.png)

**After - Pass**
```c#
public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
{
    app.UseStaticFiles();

    app.UseRouting();

    app.UseDetection();      

    app.UseEndpoints(endpoints => { endpoints.MapRazorPages(); });
}
```
![image](https://user-images.githubusercontent.com/10666633/74633413-34bc8080-5194-11ea-8f24-576dcb430e91.png)

Need to add validation that `app.UseRouting() is not in the middleware pipe before `app.UseDetection()`.

**Comment by wangkanai** [comment]
![image](https://user-images.githubusercontent.com/10666633/74633863-320e5b00-5195-11ea-96b4-1421143c5a2d.png)

**Comment by wangkanai** [comment]
Fixed with PR #355
